File: gen_scripts/python3/scripts/o10.options.py
#!/usr/bin/env python3
import sys
import getopt

USAGE = """
Usage: %s [-c] [-e] [-l] [-k] [-p] [-m] [-t] [-h|-?]

  -c  Coffee
  -e  Espresso
  -l  Latte
  -k  Machiato
  -p  Capucino
  -m  Mocha
  -t  Tea
  -h  Display this help message
  -?  Display this help message
"""

FLAGS = {
    "-c": "coffee",
    "-e": "espresso",
    "-l": "latte",
    "-k": "macchiato",
    "-p": "capucino",
    "-m": "mocha",
    "-t": "tea",
}

try:
    opts, args = getopt.getopt(sys.argv[1:], "celkpmth?")
except getopt.GetoptError:
    opts = []

for opt, _ in opts:
    if opt in ("-h", "-?"):
        print(USAGE % sys.argv[0])
        sys.exit(0)

orders = [FLAGS[opt] for opt, _ in opts if opt in FLAGS]

if not orders:
    print(USAGE % sys.argv[0], file=sys.stderr)
    sys.exit(1)

print()
print("You ordered: ")
for drink in orders:
    print("* " + drink)

File: lessons/gen_scripts/ruby/scripts/o10.options.rb
#!/usr/bin/env ruby
usage = <<END

Usage: #{$0} [-c] [-e] [-l] [-k] [-p] [-m] [-t] [-h|-?]

  -c  Coffee
  -e  Espresso
  -l  Latte
  -k  Machiato
  -p  Capucino
  -m  Mocha
  -t  Tea
  -h  Display this help message
  -?  Display this help message

END

flags = {
  "-c" => "coffee",
  "-e" => "espresso",
  "-l" => "latte",
  "-k" => "macchiato",
  "-p" => "capucino",
  "-m" => "mocha",
  "-t" => "tea",
}

orders = []
ARGV.each do |arg|
  if arg == "-h" || arg == "-?"
    print usage
    exit 0
  elsif flags.key?(arg)
    orders << flags[arg]
  end
end

if orders.empty?
  $stderr.print usage
  exit 1
end

puts ""
puts "You ordered: "
orders.each do |drink|
  puts "* #{drink}"
end
